server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -341,7 +341,9 @@ public void rollTranslogGeneration() {
     }
 
     @Override
-    public void restoreLocalCheckpointFromTranslog() {
+    public int restoreLocalHistoryFromTranslog(TranslogRecoveryRunner translogRecoveryRunner) {
+        assert false : "this should not be called";
```

Comment:
I don't understand why this throws an exception. if you have an index that is read-only and uses this engine and a primary get's promoted this should be a no-op not a UOE?

Reply:
Yes, we should make this a no-op (just like fillSeqNoGaps).